src/main/java/ContactList.java

```diff
@@ -1,70 +1,86 @@
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
+ * Class to keep track of a collection of Friend objects.
+ */
 public class ContactList {
 
     static final int DEFAULT_CAPACITY = 10;
+    static final int NOT_FOUND = -1;
 
-    private int friendCount;
+    private int size;
     private Friend[] friends;
 
-//    public ContactList() {
-//        friendCount = 0;
-//        friends = new Friend[DEFAULT_CAPACITY];
-//    }
-
     public ContactList() {
         // Call the constructor that
-        // takes an int as param
+        // takes an int as parameter
         this(DEFAULT_CAPACITY);
     }
 
     public ContactList(int capacity) {
-        friendCount = 0;
+        size = 0;
         friends = new Friend[capacity];
     }
 
     /**
-     * Add a new friend to the friends array. Will create an instance
-     * of a Friend based on parameters. If our array runs out of space
-     * we will expand capacity to manage the situation.
+     * Add a new friend to the ContactList.
      *
-     * @param firstName friends first name
-     * @param lastName  friends last name
-     * @param email     friends email address
+     * @param friend Friend object to add to the ContactList.
+     * @return True if the friend was added successfully, false otherwise.
      */
-    public void add(String firstName, String lastName, String email) {
-        // Create the Friend object
-        Friend newFriend = new Friend(firstName, lastName, email);
-
+    public boolean add(Friend friend) {
         // If we have run out of space in our array
-        // we need to deal with it
-        if (friendCount == friends.length) {
+        // we need to deal with it by making a new array
+        if (size() == friends.length) {
             expandCapacity();
         }
         // Add friend to the next available spot
-        friends[friendCount] = newFriend;
-        friendCount++;
+        friends[size] = friend;
+        size++;
+        return true;
     }
 
+    /**
+     * Private helper method for making room in the collection for new Friend objects. This method creates a new Friend
+     * array twice the size of the original (this.friends), copies over the contents of the original Friends array in
+     * order, and then assigns the friends class attribute to reference the new, bigger friends array.
+     */
     private void expandCapacity() {
         // Make a new array of twice the size of the previous
         Friend[] newFriends = new Friend[friends.length * 2];
 
-        // Copy over the contents of the friends list
-        // to the new bigger friends list
+        // Copy over the contents of the friends array
+        // to the new bigger friends array
         for (int i = 0; i < friends.length; ++i) {
             newFriends[i] = friends[i];
         }
         // Have friends now reference the new friends
         friends = newFriends;
```

Comment:
Are we avoiding `Arrays#copyOf` here?

Reply:
For better or worse, yes. Specifically doing it for the same of learning.
Perhaps later in the semester it'll change, unlike last year which used this `expandCapacity` throughout.

Reply:
All right, it's reasonable the first few times they see it/do it. No need to throw the whole of the JDK at them at once.